src/ssl.c

```diff
@@ -16623,6 +16623,91 @@ int wolfSSL_CTX_set_max_proto_version(WOLFSSL_CTX* ctx, int ver)
     return sanityCheckProtoVersion(ctx);
 }
 
+static int get_default_min_proto_version(int minDowngrade)
+{
+    int ret;
+
+    switch (minDowngrade) {
+#ifndef NO_OLD_TLS
+    #ifdef WOLFSSL_ALLOW_SSLV3
+        case SSLv3_MINOR:
+            ret = SSL3_VERSION;
+            break;
+    #endif
+    #ifdef WOLFSSL_ALLOW_TLSV10
+        case TLSv1_MINOR:
+            ret = TLS1_VERSION;
+            break;
+    #endif
+        case TLSv1_1_MINOR:
+            ret = TLS1_1_VERSION;
+            break;
+#endif
+#ifndef WOLFSSL_NO_TLS12
+        case TLSv1_2_MINOR:
+            ret = TLS1_2_VERSION;
+            break;
+#endif
+#ifdef WOLFSSL_TLS13
+        case TLSv1_3_MINOR:
+            ret = TLS1_3_VERSION;
+            break;
+#endif
+        default:
+            ret = 0;
+            break;
+    }
+
+    return ret;
+}
+
+WOLFSSL_API int wolfSSL_CTX_get_min_proto_version(WOLFSSL_CTX* ctx)
+{
+    int ret;
+
+    WOLFSSL_ENTER("wolfSSL_CTX_get_min_proto_version");
+
+    if (ctx == NULL) {
+        ret = get_default_min_proto_version(WOLFSSL_MIN_DOWNGRADE);
+    }
+    else {
+        switch (ctx->minDowngrade) {
```

Comment:
Why not call gert_default_min_proto_version with ctx->minDowngrade and if 0, call with WOLFSSL_MIN_DOWNGRADE?

Reply:
Yeah, that's cleaner. I'll change.

Reply:
Initialise ret to 0, check ctx not NULL and call function and then if ret is 0 call with define.

Reply:
Still want to call with the define when ctx is NULL.

Reply:
I think that just brings us back to what I had before, no?

Reply:
```
WOLFSSL_API int wolfSSL_CTX_get_min_proto_version(WOLFSSL_CTX* ctx)
{
    int ret = 0;

    WOLFSSL_ENTER("wolfSSL_CTX_get_min_proto_version");

    if (ctx == NULL) {
        ret = GetMinProtoVersion(WOLFSSL_MIN_DOWNGRADE);
    }
    else {
        ret = GetMinProtoVersion(ctx->minDowngrade);

        if (ret == 0) {
            ret = GetMinProtoVersion(WOLFSSL_MIN_DOWNGRADE);
        }
    }

    WOLFSSL_LEAVE("wolfSSL_CTX_get_min_proto_version", ret);

    return ret;
}
```

Reply:
How about:
WOLFSSL_API int wolfSSL_CTX_get_min_proto_version(WOLFSSL_CTX* ctx)
{
    int ret = 0;

    WOLFSSL_ENTER("wolfSSL_CTX_get_min_proto_version");

    if (ctx != NULL) {
        ret = GetMinProtoVersion(ctx->minDowngrade);
    }
    if (ret == 0) {
        ret = GetMinProtoVersion(WOLFSSL_MIN_DOWNGRADE);
    }

    WOLFSSL_LEAVE("wolfSSL_CTX_get_min_proto_version", ret);

    return ret;
}

Reply:
Looks good to me. Will re-push.